新增社群及志工頁面

Chi-Chi-Chi-Chita opened the pull request:
#387 
pop-in 目前先將資料寫死，待 API 完成再來串
pop-in 效果沒寫好，下週會研究改正
新增 Poppins 字型

Chi-Chi-Chi-Chita committed:
新增社群及志工頁面
pop-in 目前先將資料寫死，待 API 完成再來串
pop-in 效果沒寫好，下週會研究改正
新增 Poppins 字型

Chi-Chi-Chi-Chita commented:
@Tzuhui 請幫忙 Review 感謝

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+        </h2>
+        <h1>
+          我們在打造
+          </br>
```
<br> 前面多一個斜線

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+            高速資訊交流圈
+          </sapn>
+        </h1>
+        <h5 class="slogan-content">
```
這邊的語意結構，我覺得這樣可能更好一些～
```
<div class="slogan-content">
  <p>MOPCON 為締造人、知識與產業的連結，旨在實現三項目標：</p>
  <ul class="ml-3">
    <li>一、帶動南台灣高經濟產業循環</li>
    <li>二、培養高知識產值人才</li>
    <li>三、匯聚知識、人流、投資，促進創新創業機會。</li>
  </ul>
  <p>為使目標達成，MOPCON 志在打造出一座「高速資訊交流圈」！</p>
</div>
```

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+      </h2>
+      <div class="flex justify-center">
+        <button class="joinus-button" @click="fbLink">
+          <span>追蹤我們的FaceBook</span>
```
這邊也可以不用再使用 span 包住～

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+          <div class="volunteer-pop-close" @click.prevent="toggleMember ('committee')">
+            <img src="../assets/volunteerAndCommunity/close.svg" class="cursor-pointer" alt="">
+          </div>
+          <div class="volunteer-pop-card">
```
這邊我覺得可以用 max-width 去設定 Modal 的最大寬度
讓所有開啟的 Modal 大小可以一致

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+  &-card{
+    @apply mb-8;
+    &-head{
+      height:161px;
```
這邊我覺得可以不寫死高度，改用 py-8 來推出距離～

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+            </div>
+            <div class="community-pop-card-content">
+              <h5><span>\\</span>介紹</h5>
+              <h6>
```
介紹內容都可以使用文字 p 標籤就好～

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+      </h6>
+    </div>
+  </div>
+  </div>
```
多一個結尾標籤

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+      <h3 class="standard-ch">
+        我們的<span></br>行為準則</span>
+      </h3>
+      <h6 class="standard-content">
```
文字段落可以使用 p 標籤，把文字內容包住可以用 div 標籤
因為 h6 屬於標題標籤，建議還是使用在標題上哩

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+    CFSLink () {
+      window.open('https://drive.google.com/file/d/1FUuIrfc_zLPdHvVkwQ2-aUYofyvuPHso/view?usp=sharing', '_blank')
+    },
+    gdgFbLink () {
```
如果這邊接 API 後還是會使用 window.open 的話，可以試著將內容整合成一個函式。
例如：
```
linkTo (url) {
  window.open(url, '_blank')
},
```
這樣就可以重複使用～

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+      </div>
+    </div>
+    <!--社群-->
+    <div class="md:max-w-screen-xl community">
```
定義最外層寬度的部分可以直接用 .container 讓內容寬度統一

Tzuhui commented on 2020-dev/nuxt.config.js:
```diff
@@ -71,8 +71,10 @@ module.exports = {
       { rel: 'icon', type: 'image/x-icon', href: '/2020/favicon.ico' },
-      { rel: 'stylesheet', href: 'https://fonts.googleapis.com/icon?family=Material+Icons' }
+      { rel: 'stylesheet', href: 'https://fonts.googleapis.com/icon?family=Material+Icons' },
+      { rel: 'stylesheet', href: 'https://fonts.googleapis.com/css2?family=Poppins&display=swap' }
```
因為你是使用 font-weight:600，所以也要將字重加入進去才會有效果
{ rel: 'stylesheet', href: 'https://fonts.googleapis.com/css2?family=Poppins:wght@600&display=swap' }

Tzuhui commented on 2020-dev/pages/volunteerAndCommunity.vue:
```diff
@@ -0,0 +1,1147 @@
+  }
+  &-third{
+    @include btn(#651fff);
+    box-shadow: 10px 10px 0 rgba(101, 31, 255, 0.32);
```
這個可以直接用 box-shadow: 10px 10px 0 rgba(#651fff, 0.32)，我感覺比較直觀 xDD
不過你的用法也是 ok


或是要在 mixin 中加入也可以
```
@mixin btn($color, $shadow-opacity){
  box-shadow: 10px 10px 0 rgba($color, $shadow-opacity);
  ...省略已經定義的內容
}
```
```
&-second{
    @include btn(#ffcc00, 0.5)
  }
```

Tzuhui reviewed (commented):
1. 彈出的 Modal 可以用 fixed 的方式出現在畫面上
可以參考看看 Bootstrap 的 Modal
https://getbootstrap.com/docs/4.5/components/modal/
因為目前如果滾動的話，Modal 也會離開畫面
2. `<br>` 的使用再請幫忙調整一下～
3. 因為檔名就是路徑，想請你幫忙調整成 `/community`

Chi-Chi-Chi-Chita committed:
Pop Modal 及 API 1. Components 新增 Pop Modal 2. 串接 API

Chi-Chi-Chi-Chita committed:
調整顏色及微調標題位置

Chi-Chi-Chi-Chita commented:
Hi @Tzuhui 
1. 上次 code review 提到的調整都完成惹。
2. API 也完成串接惹。
3. Pop 效果改成使用 2019 年的 components 完成。
4. 顏色也修正完畢。

再麻煩你了。

Chi-Chi-Chi-Chita committed:
贊助 PPT 連結取消，改為贊助組連絡信箱。

Tzuhui commented on 2020-dev/pages/community.vue:
```diff
@@ -0,0 +1,770 @@
+        .$get('/api/2020/community')
+        .then(({ success, data, message }) => {
+          if (success) {
+            vm.communityList = data.community
```
這次只會有參與社群，所以是取 data.participant

Tzuhui commented on 2020-dev/pages/community.vue:
```diff
@@ -0,0 +1,770 @@
+        const vm = this
+        vm.communityData = []
+        vm.$axios
+          .$get(`/api/2020/community/organizer/${id}`)
```
這邊也會改成 `/api/2020/community/participant/${id}`

Tzuhui commented on 2020-dev/components/Modal/style.scss:
```diff
@@ -0,0 +1,64 @@
+.modal.fade .modal-content {
+  border-radius: 8px;
+  margin: 50px auto;
+  padding: 22px;
```
這邊的 padding 感覺可以拿掉，否則在行動版看起來會有點窄

Tzuhui commented on 2020-dev/components/Modal/style.scss:
```diff
@@ -0,0 +1,64 @@
+  overflow-y: auto;
+}
+
+.modal {
```
這邊可以加上有透明度的黑色 background-color，讓 Modal 打開時有在上層的感覺

Tzuhui commented on 2020-dev/pages/community.vue:
```diff
@@ -0,0 +1,770 @@
+          </h5>
+          <div class="logo logo-donate mx-auto" />
+          <a class="btn-third" href="mailto:sponsor@mopcon.org">
+            有興趣贊助我們
```
在行動版時可以讓文字置中～

Chi-Chi-Chi-Chita replied:
我自己剛剛也發現在 iPhone5/SE 下，寬度有點不夠，所以順手也加了 width ~

hashman replied:
讚

Tzuhui commented on 2020-dev/components/Modal/style.scss:
```diff
@@ -0,0 +1,64 @@
+/* The Modal (background) */
+.modal-open {
```
因為 scoped 的關係，這個放在這邊不會對 body 層得 .modal-open 產生效果
建議可以移動到 _base.scss 上

Tzuhui commented on 2020-dev/pages/community.vue:
```diff
@@ -0,0 +1,770 @@
+      </Modal>
+    </div>
+    <!--standard-->
+    <div class="md:max-w-screen-sm standard">
```
可以再加點 margin-bottom 推出與 footer 的距離

Chi-Chi-Chi-Chita committed:
調整
1. 2020 社群 API
2. Modal 的 background 及 padding
3. Modal 的 .modal-open 移動到 _base.scss 上
4. 推出 fooer 與上面 standard 距離

Chi-Chi-Chi-Chita commented:
Hi @Tzuhui 
感謝妳 code review 的建議，
我修改完成惹。
